[SPARK-28822][DOC][SQL]Document USE DATABASE in SQL Reference.

shivusondur opened the pull request:
<!--
Thanks for sending a pull request!  Here are some tips for you:
  1. If this is your first time, please read our contributor guidelines: https://spark.apache.org/contributing.html
  2. Ensure you have added or run the appropriate tests for your PR: https://spark.apache.org/developer-tools.html
  3. If the PR is unfinished, add '[WIP]' in your PR title, e.g., '[WIP][SPARK-XXXX] Your PR title ...'.
  4. Be sure to keep the PR description updated to reflect all changes.
  5. Please write your PR title to summarize what this PR proposes.
  6. If possible, provide a concise example to reproduce the issue for a faster review.
-->

### What changes were proposed in this pull request?
Added document reference for USE databse sql command
<!--
Please clarify what changes you are proposing. The purpose of this section is to outline the changes and how this PR fixes the issue. 
If possible, please consider writing useful notes for better and faster reviews in your PR. See the examples below.
  1. If you refactor some codes with changing classes, showing the class hierarchy will help reviewers.
  2. If you fix some SQL features, you can provide some references of other DBMSes.
  3. If there is design documentation, please add the link.
  4. If there is a discussion in the mailing list, please add the link.
-->


### Why are the changes needed?
For USE database command usage
<!--
Please clarify why the changes are needed. For instance,
  1. If you propose a new API, clarify the use case for a new API.
  2. If you fix a bug, you can clarify why it is a bug.
-->


### Does this PR introduce any user-facing change?
It is adding the USE database sql command refernce information in the doc
<!--
If yes, please clarify the previous behavior and the change this PR proposes - provide the console output, description and/or an example to show the behavior difference if possible.
If no, write 'No'.
-->


### How was this patch tested?
Attached the test snap
![image](https://user-images.githubusercontent.com/7912929/65170499-7242a380-da66-11e9-819c-76df62c86c5a.png)


<!--
If tests were added, say they were added here. Please make sure to add some test cases that check the changes thoroughly including negative and positive cases if possible.
If it was tested in a way different from regular unit tests, please clarify how you tested step by step, ideally copy and paste-able, so that other reviewers can test and check, and descendants can verify in the future.
If tests were not added, please describe why they were not added and/or why it was difficult to add.
-->

shivusondur committed:
Updated the document details

AmplabJenkins commented:
Can one of the admins verify this patch?

AmplabJenkins commented:
Can one of the admins verify this patch?

AmplabJenkins commented:
Can one of the admins verify this patch?

amanomer commented on docs/sql-ref-syntax-qry-select-usedb.md:
```diff
@@ -0,0 +1,41 @@
+
+### Example
+{% highlight sql %}
+USE userdb
```
;

amanomer commented on docs/sql-ref-syntax-qry-select-usedb.md:
```diff
@@ -0,0 +1,41 @@
+
+ **dbname**
+
+ dbname is used as current database, If the provided database does not exist, an exception is thrown.
```
I think dot(.) should come instead of comma(,) after 'current database'?

amanomer commented on docs/sql-ref-syntax-qry-select-usedb.md:
```diff
@@ -0,0 +1,41 @@
+### Example
+{% highlight sql %}
+USE userdb
+{% endhighlight %}
```
Add related statements like SHOW DATABASE and CREATE DATABASE.

shivusondur replied:
done

amanomer commented on docs/sql-ref-syntax-qry-select-usedb.md:
```diff
@@ -0,0 +1,41 @@
+
+### Description
+Set the provided databse as current database. All subsequent commands that do not explicitly
+specify a database will use this one. The default current database is default
```
This seems similar to the databricks USE database definition (https://docs.databricks.com/spark/latest/spark-sql/language-manual/use-database.html).
Can we re-word this to "Set the provided database as current database. All subsequent queries will try to resolve table names to this database. The default database is `default`." ?
cc @gatorsmile  @srowen  @dilipbiswal

shivusondur replied:
done

shivusondur committed:
Updated the document details

amanomer commented on docs/sql-ref-syntax-qry-select-usedb.md:
```diff
@@ -0,0 +1,41 @@
+
+### Description
+Set the provided database as current database. All subsequent queries will resolve
+table names to this database if database name is not provided. The default database name is 'default'
```
Use backtick( ` ) instead of single quote( ' ).

shivusondur committed:
Handled PR according to #25523 PR which is already merged.
to make uniform accross all docs

shivusondur commented:
@dilipbiswal @gatorsmile
Updated the PR according to #25525 and #25523 PR's.
Updated the ScreenShot after creating the document from Jekyll build

Please review and kindly check for merging.

AmplabJenkins commented:
Can one of the admins verify this patch?

dilipbiswal commented on docs/sql-ref-syntax-qry-select-usedb.md:
```diff
@@ -0,0 +1,59 @@
+### Parameter
+
+<dl>
+  <dt><code><em>{FROM|IN} database_name</em></code></dt>
```
@shivusondur Shouldn't { FROM | IN } be in the main syntax diagram if its mentioned as a parameter here ?

shivusondur replied:
Handled

dilipbiswal commented on docs/sql-ref-syntax-qry-select-usedb.md:
```diff
@@ -0,0 +1,59 @@
+---
+
+### Description
+Set the provided database as current database. All subsequent queries that do not explicitly
```
@shivusondur How about 
```
`USE` statement is used to set a current database. After a current database is set, the unqualified database artifacts such as tables, functions  and views that are referenced by SQLs are resolved from the current database. 
```

shivusondur replied:
Handled

shivusondur committed:
Handled the comments

shivusondur commented:
@dilipbiswal 
Thanks for your time for review
I have handled all your comments and updated the screen shot.
please check

dilipbiswal commented:
LGTM
cc @gatorsmile

gatorsmile committed:
Update sql-ref-syntax-qry-select-usedb.md

gatorsmile commented:
ok to test

SparkQA commented:
**[Test build #111013 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/111013/testReport)** for PR 25572 at commit [`e370b13`](https://github.com/apache/spark/commit/e370b13c1962ef44ea221e4c25b1d43458c35838).

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder-K8s/16112/
Test PASSed.

SparkQA commented:
**[Test build #111014 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/111014/testReport)** for PR 25572 at commit [`e370b13`](https://github.com/apache/spark/commit/e370b13c1962ef44ea221e4c25b1d43458c35838).

SparkQA commented:
**[Test build #111013 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/111013/testReport)** for PR 25572 at commit [`e370b13`](https://github.com/apache/spark/commit/e370b13c1962ef44ea221e4c25b1d43458c35838).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/111013/
Test PASSed.

SparkQA commented:
**[Test build #111014 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/111014/testReport)** for PR 25572 at commit [`e370b13`](https://github.com/apache/spark/commit/e370b13c1962ef44ea221e4c25b1d43458c35838).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/111014/
Test PASSed.

gatorsmile commented:
Thanks! Merged to master.

dilipbiswal commented:
@gatorsmile Thank you very much for fixing and merging.
